blockchain/chain.go

```diff
@@ -772,9 +790,11 @@ func (b *BlockChain) disconnectBlock(node *blockNode, block *btcutil.Block, view
 		return err
 	}
 
-	// Prune fully spent entries and mark all entries in the view unmodified
-	// now that the modifications have been committed to the database.
-	view.commit()
+	// Commit all modifications made to the view into the utxo state.  This also
+	// prunes these changes from the view.
+	b.stateLock.Lock()
+	b.utxoCache.Commit(view)
```

Comment:
Why not catch the error here as well? 😃 
```suggestion
	if err := b.utxoCache.Commit(view); err != nil {
		log.Errorf("error committing block %s(%d) to utxo cache: %s", block.Hash(), block.Height(), err.Error())
	}

```